### ISSUE TITLE
CMake issues building on Windows + VS 2015

### ISSUE THREAD

**Issue by danginsburg** [KhronosGroup/Vulkan-ValidationLayers#113]
I had some struggles getting this repository to build since separating the repos and I figured it was worth sharing the problems since some of them may be worth fixing.

1. As far as I can figure out, it's not possible to use cmake-gui any more since command-line options are required to set GLSLANG_REPO_ROOT and I can't see any way to set it in the GUI since it's not a variable that's exposed.  Attempting to configure in cmake-gui I get:

ERRORMust define GLSLANG_REPO_ROOT -- see BUILD.md

I wouldn't be surprised if this impacts ccmake on Linux as well? (haven't tried)  This used to work in the old combined repo.

2. As far as I can find in BUILD.md, there was nothing to tell me that I needed to build glslang in-tree with folder name build (64-bit) or build32 (32-bit).  This should be documented and/or configurable.

3. I happened to build glslang RelWithDebInfo and the libraries were not found.  Digging through CMakeLists.txt I discovered only Release/Debug builds will be found.  This should also probably be documented.

4. I had to explicitly set LIBVK in addition to LOADER_REPO_ROOT.  Is that expected?  It wasn't documented.  This is the cmake command-line I'm using:

```
C:\Users\dang\src\Vulkan-ValidationLayers\build>cmake -DLOADER_REPO_ROOT=C:\VulkanSDK\1.1.73.0 -DGLSLANG_REPO_ROOT=c:\users\dang\src\glslang -DLIBVK=c:\VulkanSDK\1.1.73.0\Lib\vulkan-1.lib -G "Visual Studio 14 2015 Win64" ..
```

**Comment by mark-lunarg** [comment]
Thanks Dan.  Some of this we're working on now, but much of this is new and very useful.  A question:  we were considering requiring that glslang be built with an install target (per its instructions), and asking the user to point to the install directory instead of poking around in the repo's build artifacts -- does that seem reasonable?  This would help to avoid the build-directory naming mess.

**Comment by danginsburg** [comment]
> A question: we were considering requiring that glslang be built with an install target (per its instructions), and asking the user to point to the install directory instead of poking around in the repo's build artifacts -- does that seem reasonable?

Yes, that was actually what I figured it would do, so I started by having it point to the glslang install dir but it didn't work.  That seems much more sane.  The instructions will just need to include setting CMAKE_INSTALL_PREFIX for glslang because by default on Windows it's to c:\program files and most people probably don't want to do that.  It also will need to tell them to build the INSTALL project.  I'm not sure if that happens by default, I think you might need to build it explicitly.
